src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -63,6 +64,10 @@ public function collect(Request $request, Response $response, \Exception $except
             if ('_route' === $key) {
                 $route = is_object($value) ? $value->getPath() : $value;
             }
+
+            if ('_route_params' === $key && is_array($value)) {
+                $routeParams = $value;
```

Comment:
I agree with @wouterj , here is the patch (fixing session data display meanwhile):
```patch
--- a/src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php
+++ b/src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php
@@ -56,6 +56,10 @@ class RequestDataCollector extends DataCollector implements EventSubscriberInter
         foreach ($request->attributes->all() as $key => $value) {
             if ('_route' === $key && is_object($value)) {
                 $attributes[$key] = $this->cloneVar($value->getPath());
+            } elseif ('_route_params' === $key) {
+                foreach ($value as $k => $v) {
+                    $attributes[$key][$k] = $this->cloneVar($v);
+                }
             } else {
                 $attributes[$key] = $this->cloneVar($value);
             }
@@ -83,7 +87,9 @@ class RequestDataCollector extends DataCollector implements EventSubscriberInter
                 $sessionMetadata['Created'] = date(DATE_RFC822, $session->getMetadataBag()->getCreated());
                 $sessionMetadata['Last used'] = date(DATE_RFC822, $session->getMetadataBag()->getLastUsed());
                 $sessionMetadata['Lifetime'] = $session->getMetadataBag()->getLifetime();
-                $sessionAttributes = $session->all();
+                foreach ($session->all() as $k => $v) {
+                    $sessionAttributes[$k] = $this->cloneVar($v);
+                }
                 $flashes = $session->getFlashBag()->peekAll();
             }
         }
@@ -264,7 +270,7 @@ class RequestDataCollector extends DataCollector implements EventSubscriberInter
      */
     public function getRouteParams()
     {
-        return isset($this->data['request_attributes']['_route_params']) ? $this->data['request_attributes']['_route_params'] : $this->cloneVar(array());
+        return isset($this->data['request_attributes']['_route_params']) ? $this->data['request_attributes']['_route_params'] : array();
     }
 
     /**
```